documentation/staging/content/faq/external-clients.md

```diff
@@ -27,15 +27,98 @@ description: "This FAQ describes approaches for giving external WebLogic clients
-
-  * [Load balancer tunneling](#load-balancer-tunneling) (preferred)
-  * [Kubernetes `NodePorts`](#kubernetes-nodeports)
-
-Conversely, if giving a Kubernetes hosted WebLogic Server access to external WebLogic Server EJBs, JMS, or JTA, then consider the following:
-
-  * You may need to [enable unknown host access](#enabling-unknown-host-access) on the external WebLogic Servers.
-  * Plus, if the target server can be accessed only through a load balancer using HTTP:
+If a WebLogic EJB or JMS resource is located in the same
+Kubernetes namespace as an application that calls the resource, then:
+
+- No additional configuration steps are needed.
+
+- The URL that the application specifies depends on both the location
+  of the application and the location of its target resource.
+  - If the application is running in a WebLogic Server JVM, and the
+    JVM hosts the target resource or the JVM
+    is located within the same WebLogic cluster
+    as the target resource, then the application
+    can simply specify the JNDI name of the resource and
+    _it should not specify a URL_.
+  - If the application is running outside of one of the above locations,
+    but is still in the same Kubernetes namespace as the target resource,
+    then, in addition to a JNDI name, the application must also specify a `t3`
+    or `t3s` URL that includes the DNS name of the target
+    resource's Kubernetes ClusterIP service. To see the service DNS names
+    for a deployed domain, run `kubectl -n MYNS get services`.
+    Example URLs:
+    - For a target resource that's hosted anywhere
+      in WebLogic cluster `mycluster`
+      with domain UID `myuid` where the cluster is listening
+      on port `7001`: `t3://myuid-cluster-mycluster:7001`.
+    - For a target resource that's hosted in a non-clustered
+      WebLogic Server `myserver` that is part
+      of domain with a domain UID `myuid` where the server
+      is listening on port `7001`: `t3://myuid-myserver:7001`.
+
+If a WebLogic EJB or JMS resource is located in the same
+Kubernetes cluster as an application that calls the resource, but
+the application and resource are in different Kubernetes namespaces, then:
+
+- The URL used by the application can begin with `t3` or `t3s`, and the
+  DNS address in the URL must be fully specified to differentiate the namespace.
+  - For example,
+    if the target resource is in WebLogic cluster `mycluster` with
+    domain UID `myuid` in namespace `myns`, and the cluster is
+    listening on port `8001`, then the cluster service name 
+    that is generated by the operator will
+    be `myuid-cluster-mycluster` and the full URL will be
+    `t3://myuid-cluster-mycluster.myns.service.cluster.local:8001`.
+
+- If the applications run within a WebLogic
+  Server JVM, then the WebLogic Servers that host the target
+  EJB or JMS resources need to be configured to
+  [enable unknown host access](#enabling-unknown-host-access).
```

Comment:
We may not need to enable unknown host access if the resource and application are in the same k8s cluster, and fully decorated DNS address is used in the public address.

Reply:
A customer recently told us on Slack that they needed to do this. I'll add the word "may" before need.

Reply:
I've updated and moved the bullet.